crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/class_info.rs

```diff
@@ -296,14 +300,53 @@ pub struct ClassInfo {
 pub fn get_class_info(class_name: &str, sort_config: &SortConfig) -> Option<ClassInfo> {
     let utility_data = tokenize_class(class_name)?;
     let utility_info = get_utility_info(sort_config.utilities, &utility_data.utility);
+
     if let Some(utility_info) = utility_info {
         return Some(ClassInfo {
             text: class_name.to_string(),
             variant_weight: if utility_data.variants.is_empty() {
                 None
             } else {
-                // TODO: return None if there is an unknown variant.
-                Some(0) // TODO: actually compute variant weight
+                let known_variants = utility_data.variants.iter().any(|variant| {
+                    sort_config
```

Comment:
Should probably just be a hashmap access of sorts for speed.

Reply:
I agree on this, I went with this to not revolutionize everything
I feel that a hashmap structure would then have a value about arbitrariety and could not be a simple `HashMap<string, BitVec>`, am I right?

Reply:
My hope regarding arbitrary variants is that we won't need to make a distinction between variants that support it and variants that don't, and we'll be able to just take a "starts with" kind of approach for simplicity, since the only significant trade-off there is the possibility of false positives which is very minor. This is similar to how utilities are handled and it seems to work well enough.

Reply:
Probably `endsWith`?
I would then have to generate for example `has-[]` instead of `has` that is currently generated (same as `group-[]` instead of group)
What do you think?

Reply:
Went reading Tailwind docs more and testing the plugin more about this.
My previous comment doesn't apply at all 😄 
But wanting to point out that `group-[:checked]` keeps `group` weight when sorting, but `group-hover` has its own weight. From the generated Variants there's also `group-aria-busy` for example that has its own weight. By doing `starts_with`, since it comes after `group` in the `Vec<Variants>` we would have a false positive when the iterating step is on `group`

I think we should do the matching logic this way:
- Does the variant on the classes endsWith "]" AND its prefix (as a whole though) is a recognized variant? if yes, then it's a known variant and we can compute weights
- Keeping the same logic on the exact match between variants names

I've also noticed that `[&nth-child(2)]:py-4` are put right before everything, before `custom classes` or `components layer` utility classes, while the plugin puts them right after every recognized variant and orders them lexicographically, so that's another thing to tackle.

Reply:
Check out how I handle this situation in utilities. You'll see that there's logic that gives priority to longer matches over shorter ones. I think that should probably work here too and be quite efficient.

https://github.com/lutaok/biome/blob/b797e786905b96611a262025000c3689b28fdc4d/crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/class_info.rs#L140-L150

Reply:
Thanks, yes I copied your function because the structure of the variants is somewhat different, but I wonder if we could make it reusable enough